src/upgrade/migrator.js

```diff
@@ -0,0 +1,88 @@
+/* Copyright (C) 2016 NooBaa */
+'use strict';
+
+const _ = require('lodash');
+const P = require('../util/promise');
+
+/**
+ *
+ * Migrator
+ *
+ * taking two db's and move all the given collections between them
+ *
+ */
+class Migrator {
+
+    /**
+     * @param {Object[]} collection_list
+     * @param {nb.DBClient} from_client
+     * @param {nb.DBClient} to_client
+     * @param {number} [batch_size]
+     */
+    constructor(from_client, to_client, collection_list, batch_size) {
+        this.collection_list = collection_list;
+        this.from_client = from_client;
+        this.to_client = to_client;
+        this.batch_size = batch_size || 100;
+    }
+
+    async migrate_db() {
+        this.to_client.define_collection({ name: 'migrate_status' });
+        for (const collection of this.collection_list) {
+            await this.from_client.define_collection(_.omit(collection, 'db_indexes'));
+            await this.to_client.define_collection(collection);
+        }
+        await this.from_client.connect(true);
+        await this.to_client.connect();
+        this.upgrade_table = this.to_client.collection('migrate_status');
+        this.migrate_status = await this.upgrade_table.findOne({});
+        if (!this.migrate_status) {
+            this.migrate_status = {
+                _id: this.to_client.new_object_id(),
+                collection_index: 0,
```

Comment:
add more info to help with debugging - collection_name, write_date, total_size, num_inserted, etc.

Reply:
done